Golem/Yagna/YagnaApi.cs

```diff
@@ -0,0 +1,276 @@
+
+        public async Task<List<Payment>> GetPayments(CancellationToken token = default)
+        {
+            List<Payment>? payments = null;
+            try
+            {
+                var paymentsResponse = await _httpClient.GetAsync($"/payment-api/v1/payments", token);
+
+                if (paymentsResponse.IsSuccessStatusCode)
+                {
+                    var result = await paymentsResponse.Content.ReadAsStringAsync();
+                    if (result != null)
+                    {
+                        payments = JsonSerializer.Deserialize<List<Payment>>(result, _serializerOptions);
+                        _logger.LogDebug("payments {0}", payments != null ? payments.SelectMany(x => x.AgreementPayments.Select(y => y.AgreementId + ": " + y.Amount)).ToList() : "(null)");
+                    }
+                }
+            }
+            catch (Exception e)
+            {
+                _logger.LogError("GetPayments error: {msg}", e.Message);
+            }
+
+            return payments ?? new List<Payment>();
+        }
+
+        public async Task<List<InvoiceEvent>> GetInvoiceEvents(DateTime since, CancellationToken token = default)
+        {
+            const int timeout = 10;
+
+            var url = $"/payment-api/v1/invoiceEvents?timeout[timeout]={timeout}&afterTimestamp={since.ToUniversalTime().ToString("o", CultureInfo.InvariantCulture)}";
+            using var requestMessage = new HttpRequestMessage(HttpMethod.Get, url);
+
+            requestMessage.Headers.Add("X-Requestor-Events", string.Join(',', _monitorEventTypes));
+            requestMessage.Headers.Add("X-Provider-Events", string.Join(',', _monitorEventTypes));
+
+            try
+            {
+                var invoiceEventsResponse = await _httpClient.SendAsync(requestMessage, token);
+                if (invoiceEventsResponse.IsSuccessStatusCode)
+                {
+                    var result = await invoiceEventsResponse.Content.ReadAsStringAsync();
+                    if (result != null)
+                    {
+                        _logger.LogDebug("InvoiceEvent: {0}", result);
+                        var invoiceEvents = JsonSerializer.Deserialize<List<InvoiceEvent>>(result, _serializerOptions);
+                        return invoiceEvents ?? new List<InvoiceEvent>();
+                    }
+                }
+                else
+                {
+                    _logger.LogError("Got invoiceEvents {0}", invoiceEventsResponse);
+                    await Task.Delay(TimeSpan.FromSeconds(1), token);
+                }
+            }
+            catch (TaskCanceledException)
+            {
+                _logger.LogDebug("Payment loop cancelled");
+            }
+            catch (Exception e)
+            {
+                _logger.LogError(e, "Payment request failure");
+                await Task.Delay(TimeSpan.FromSeconds(5), token);
+            }
+
+            return new List<InvoiceEvent>();
+        }
```

Comment:
Could this be simplified to something like `RestCall<List<InvoiceEvent>>`?
I understand that additional headers might be a problem. Maybe you need to create another variant of `RestCall` that allows to pass additional things.